server/index.js

```diff
@@ -97,13 +98,45 @@ app.get('/api/search/:endpoint', (req, res, next) => {
         const albumList = data.map(({ title, cover, artist: { name } }) => ({ title, cover, name }));
         res.status(201).json(albumList);
       } else if (endpoint === 'track') {
-        // eslint-disable-next-line camelcase
-        const trackList = data.map(({ title, explicit_lyrics, artist: { name }, album: { cover } }) => ({ title, explicit_lyrics, name, cover }));
+        const trackList = data.map(
+          // eslint-disable-next-line camelcase
+          ({ id, title, explicit_lyrics, artist: { id: artistId, name: artistName }, album: { id: albumId, cover: albumCover } }) => ({ id, title, explicit_lyrics, artistId, artistName, albumId, albumCover })
+        );
         res.status(201).json(trackList);
       }
     });
 });
 
+app.use(authorizationMiddleware);
+
+app.post('/api/save/library', (req, res, next) => {
+  const { userId } = req.user;
+  const { id, title, artistId, albumId } = req.body;
+  const sql = `
+    insert into "tracks" ("trackId", "title", "artistId", "albumId")
+    values ($1, $2, $3, $4)
+    returning "trackId"
+  `;
```

Comment:
Here we'll need to account for different users attempting to save the same track. to prevent duplicates, we have a primary key on the track id. That's good. However, this insert could cause a promise rejection because a 2nd user could insert a duplicate track. Fortunately, we can `do nothing` if there's a conflict.

```sql
    insert into "tracks" ("trackId", "title", "artistId", "albumId")
    values ($1, $2, $3, $4)
    on conflict ("trackId")
    do nothing
    returning "trackId"
```

This will require an adjustment to the `params` on line 129, in case we don't get back the freshly inserted row from the tracks table. It's ok though, because we have `id` from the `req.body`, so we can just use that instead.